brave/brave-core
# Rotate search promotion for DDG users

fix https://github.com/brave/brave-browser/issues/38811

Rotate omnibox search promotion banner for DDG every minutes.

There are 4 types of search promotion for DDG users
![Screenshot from 2024-06-07 09-34-33](https://github.com/brave/brave-core/assets/6786187/74b620c0-cf4b-4557-9d1e-eae245d7131f)
![Screenshot from 2024-06-07 09-29-13](https://github.com/brave/brave-core/assets/6786187/42b0b596-1fea-4026-9ff6-3ffa690105bf)
![Screenshot from 2024-06-07 09-31-23](https://github.com/brave/brave-core/assets/6786187/803f48fb-9b00-4795-a4c7-a16281e92bac)
![Screenshot from 2024-06-07 09-33-27](https://github.com/brave/brave-core/assets/6786187/be9ae5c6-66a4-4be1-85c4-d659eafe973e)



Resolves



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

`npm run test brave_unit_tests -- --filter=braveBraveSearchConversion*`

This promotion is disabled by default and will be controlled via griffin.
To test easily w/o griffin, we could use `--enable-features=XXX` cmd args

1. Launch brave with `--enable-features=BraveSearchOmniboxDDGBanner`
2. Set DDG as a default search provider
3. Type anything in omnibox and check DDG promotion entry is shown at last
4. Wait 1mins at least to see another DDG promotion
5. type anything and check different DDG promotion is visible
6. Repeat 4-5 several times to see 4 different types of promotion is shown
7. Set Google as a default search provider
8. Type anything in omnibox and check there is no promotion
9. Set Brave as a default search provider
10. Type anything in omnibox and check there is no promotion

As this promotion is controlled with new griffin study, user could get existing search promotion study togeter
When DDG promotion and existing promotion are enabled both,
user will see DDG promotion if current search provider is DDG.
And user will see existing promotion if current search provider is non-brave search.
To test easily this scenario, we could use `--enable-features=XXX` cmd args also.

1. Launch brave with `--enable-features=BraveSearchOmniboxBanner:banner_type/type_B,BraveSearchOmniboxDDGBanner`
2. Set DDG as a default search provider
3. Type anything in omnibox and check DDG promotion is shown
4. Set Google as a default search provider
5. Type anything in omnibox and check previous promotion is shown
6. Set brave and check there is no promotion
